test/over_react_redux/connect_test.dart

```diff
@@ -0,0 +1,444 @@
+      });
+    });
+
+    group('Options', () {
+      group('areOwnPropsEqual', (){
+        test('', () {
+          ConnectedCounter = connect<CounterState, CounterProps>(
+            areOwnPropsEqual: expectAsync2((next, prev) {
+              expect(next, isA<CounterProps>());
+              expect(prev, isA<CounterProps>());
+              expect(next.id, 'test');
+              expect(prev.id, 'test2');
+              return true;
+            }),
+          )(Counter);
+
+          var whatever = connectOptions.areOwnPropsEqual(
+            JsBackedMap.from({'id':'test'}).jsObject,
+            JsBackedMap.from({'id':'test2'}).jsObject
+          );
+
+          expect(whatever, true);
+        });
+      });
+
+      group('areStatePropsEqual', (){
+        test('', () async {
+          List<String> methodsCalled = [];
+          ConnectedCounter = connect<CounterState, CounterProps>(
+            mapStateToProps: (state) {
+              methodsCalled.add('mapStateToProps');
+              return Counter()..currentCount = state.count;
+            },
+            areStatePropsEqual: (next, prev) {
+              expect(next, isA<CounterProps>());
+              expect(prev, isA<CounterProps>());
+              expect(next.currentCount, 1);
+              methodsCalled.add('areStatePropsEqual');
+              // Force it to always be true, meaing it shouldnt re-render if they change.
+              return true;
+            },
+            forwardRef: true,
+          )(Counter);
+
+          jacket = mount(
+            (ReduxProvider()..store = store1)(
+              (ConnectedCounter()..ref = (ref){ counterRef = ref; }..currentCount = 0)('test'),
+            ),
+          );
+          expect(methodsCalled, ['mapStateToProps']);
+          methodsCalled.clear();
+
+          var dispatchButton = getByTestId(jacket.getInstance(), 'button-increment');
+          click(dispatchButton);
+
+          // wait for the next tick for the async dispatch to propagate
+          await Future(() {});
+
+          // store.state.count should be 1 but does not re-render due to override in `areStatePropsEqual`
+
+          expect(methodsCalled, ['mapStateToProps', 'areStatePropsEqual']);
+          expect(jacket.getNode().innerHtml, contains('Count: 0'));
+        });
+      });
+
+      group('areMergedPropsEqual', (){
+        test('', () {
+          ConnectedCounter = connect<CounterState, CounterProps>(
+            areMergedPropsEqual: expectAsync2((next, prev) {
+              expect(next, isA<CounterProps>());
+              expect(prev, isA<CounterProps>());
+              expect(next.id, 'test');
+              expect(prev.id, 'test2');
+              return true;
+            }),
+          )(Counter);
+
+          var whatever = connectOptions.areMergedPropsEqual(
+            JsBackedMap.from({'id':'test'}).jsObject,
+            JsBackedMap.from({'id':'test2'}).jsObject
+          );
+
+          expect(whatever, true);
```

Comment:
#nit (same as above)
```suggestion
          expect(whatever, isTrue);
```